server/controllers/admin/nationalOccupancyController.ts

```diff
@@ -33,40 +44,78 @@ export default class NationalOccupancyController {
         query,
       } = req
 
+      let { fromDate } = query as { fromDate: string }
+
       const sessionData = this.formData.get(defaultSessionKey, req.session) || {}
-      const combinedQuery = { ...sessionData, ...query }
+
+      if (query.arrivalDate !== undefined) {
+        Object.assign(sessionData, query, {
+          apCriteria: makeArrayOfType(query.apCriteria),
+          roomCriteria: makeArrayOfType(query.roomCriteria),
+        })
+        fromDate = DateFormats.datepickerInputToIsoString(query.arrivalDate as string)
+      }
 
       const {
-        postcode,
+        postcode: rawPostcode,
         arrivalDate: rawArrivalDate,
         apArea,
         apType,
         roomCriteria,
         apCriteria,
-      } = combinedQuery as NationalSpaceSearchFormData
+      } = sessionData as NationalSpaceSearchFormData
 
       const errors: Record<string, string> = {}
 
       const slashToday = DateFormats.isoDateToUIDate(DateFormats.dateObjToIsoDate(new Date()), { format: 'datePicker' })
 
       const cruManagementAreas = await this.cruManagementAreaService.getCruManagementAreas(token)
-      const arrivalDate = rawArrivalDate || slashToday
+      const postcode = rawPostcode && rawPostcode.toUpperCase()
 
-      if (!isoDateIsValid(DateFormats.datepickerInputToIsoString(arrivalDate as string))) {
+      if (
+        query.arrivalDate !== undefined &&
+        !isoDateIsValid(DateFormats.datepickerInputToIsoString(rawArrivalDate as string))
+      ) {
         errors.arrivalDate = 'Enter a valid arrival date'
       }
+
       if (postcode && !validPostcodeArea(postcode)) {
         errors.postcode = 'Enter a valid postcode area'
       }
 
-      if (!errors.length) {
-        await this.formData.update(defaultSessionKey, req.session, combinedQuery)
+      if (!Object.keys(errors).length) {
+        await this.formData.update(defaultSessionKey, req.session, sessionData)
       }
 
+      let capacity: Cas1NationalOccupancy
+
+      if (fromDate && !Object.keys(errors).length) {
+        try {
+          capacity = await this.premisesService.getMultipleCapacity(token, {
+            cruManagementAreaIds: expandManagementArea(cruManagementAreas, apArea),
+            fromDate,
+            postcodeArea: postcode,
+            premisesCharacteristics: makeArrayOfType<Cas1SpaceCharacteristic>(apCriteria || []),
+            roomCharacteristics: makeArrayOfType<Cas1SpaceCharacteristic>(roomCriteria || []),
+          })
+
+          capacity.premises.forEach(premises => {
+            premises.distanceInMiles = roundNumber(premises.distanceInMiles, 1)
+          })
```

Comment:
This should be done inside `processCapacity`. I'm doing that but won't push until you've finished reviewing.

Reply:
I think I've just now left a comment about this -- don't think we need `roundNumber` given we can do the rounding where a string will be enough!